services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/ApplicationConfig.java

```diff
@@ -198,8 +198,8 @@ public ServiceConfigProperties restProperties() {
     @Bean
     @Scope("prototype")
     @ConditionalOnBean(DeviceManagementService.class)
-    public DeviceManagementHttpEndpoint registrationHttpEndpoint() {
-        return new DeviceManagementHttpEndpoint(vertx());
+    public AbstractDeviceManagementHttpEndpoint registrationHttpEndpoint() {
```

Comment:
is it necessary to return an `AbstractDeviceManagementHttpEndpoint` instead of just a `DeviceManagementHttpEndpoint` ?

Reply:
It is. These are not related by any means, the 'AbstractDeviceManagementHttpEndpoint' is completely new, so that 'DeviceManagementHttpEndpoint' could be deprecated.